bson/src/main/org/bson/Vector.java

```diff
@@ -0,0 +1,268 @@
+        if (this.vectorType != Dtype.INT8) {
+            throw new IllegalStateException("Vector is not INT8");
+        }
+        return assertNotNull(vectorData);
+    }
+
+    /**
+     * Returns the {@link Dtype#FLOAT32} vector data as a float array.
+     *
+     * <p> This method is used to retrieve the underlying float array representing the {@link Dtype#FLOAT32} vector, where each float
+     * represents an element of a vector.</p>
+     *
+     * @return the float array representing the FLOAT32 vector.
+     * @throws IllegalStateException if this vector is not of type {@link Dtype#FLOAT32}. Use {@link #getDataType()} to check the vector
+     *                               type before calling this method.
+     */
+    public float[] asFloatVectorData() {
+        if (this.vectorType != Dtype.FLOAT32) {
+            throw new IllegalStateException("Vector is not FLOAT32");
+        }
+
+        return assertNotNull(floatVectorData);
+    }
+
+    /**
+     * Returns the padding value for this vector.
+     *
+     * <p>Padding refers to the number of least-significant bits in the final byte that are ignored when retrieving the vector data, as not
+     * all {@link Dtype}'s have a bit length equal to a multiple of 8, and hence do not fit squarely into a certain number of bytes.</p>
+     *
+     * @return the padding value (between 0 and 7).
+     */
+    public byte getPadding() {
+        return this.padding;
+    }
+
+
+    /**
+     * Returns {@link Dtype} of the vector.
+     *
+     * @return the data type of the vector.
+     */
+    public Dtype getDataType() {
+        return this.vectorType;
+    }
+
+
+    @Override
+    public String toString() {
+        return "Vector{"
+                + "padding=" + padding + ", "
+                + "vectorData=" + (vectorData == null ? Arrays.toString(floatVectorData) : Arrays.toString(vectorData))
+                + ", vectorType=" + vectorType
+                + '}';
+    }
+
+    @Override
+    public boolean equals(final Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (!(o instanceof Vector)) {
+            return false;
+        }
+
+        Vector vector = (Vector) o;
+        return padding == vector.padding && Arrays.equals(vectorData, vector.vectorData)
+                && Arrays.equals(floatVectorData, vector.floatVectorData) && vectorType == vector.vectorType;
+    }
+
+    @Override
+    public int hashCode() {
+        int result = padding;
+        result = 31 * result + Arrays.hashCode(vectorData);
+        result = 31 * result + Arrays.hashCode(floatVectorData);
+        result = 31 * result + Objects.hashCode(vectorType);
+        return result;
+    }
+
+    /**
+     * Represents the data type (dtype) of a vector.
+     * <p>
+     * Each dtype determines how the data in the vector is stored, including how many bits are used to represent each element
+     * in the vector.
+     */
+    public enum Dtype {
```

Comment:
How about just call it "Type".  The "D" isn't going to help anyone, and might confuse people.  Also, we generally avoid abbreviations unless they are well-known terms of art, e.g. "html"

Reply:
Good point. 'DType' originates from NumPy, but I agree we should avoid abbreviations unless they're well-known. Many libraries and frameworks (for example TensorFlow and DL4J) use 'DataType' [DL4J DataType](https://github.com/deeplearning4j/deeplearning4j/blob/master/nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/buffer/DataType.java)). Using 'DataType' for the enum could maintain consistency and make the terminology clearer.

I'm open to naming it `Type `as well. Would `DataType` work, or do you prefer `Type`?

Reply:
DataType sgtm

Reply:
Renamed. Thanks!